evalml/tests/automl_tests/test_automl.py

```diff
@@ -5363,3 +5363,83 @@ def test_automl_respects_iterative_pipeline_order(X_y_binary, AutoMLTestEnv):
         searched_model_families,
         key=lambda model_family: _ESTIMATOR_FAMILY_ORDER.index(model_family),
     )
+
+
+def test_get_ensembler_input_pipelines(X_y_binary, AutoMLTestEnv):
+    X, y = X_y_binary
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_iterations=_get_first_stacked_classifier_no(),
+        objective="Log Loss Binary",
+        ensembling=True,
+    )
+
+    score_side_effect = [
+        {"Log Loss Binary": score}
+        for score in np.arange(
+            0, -1 * automl.max_iterations * automl.data_splitter.get_n_splits(), -0.1
+        )
+    ]  # Decreases with each call
+
+    test_env = AutoMLTestEnv("binary")
+    with test_env.test_context(mock_score_side_effect=score_side_effect):
+        automl.search()
+
+    best_pipeline_ids = [
+        pipeline["id"]
+        for pipeline in list(automl._automl_algorithm._best_pipeline_info.values())
+    ]
+    best_pipeline_ids.sort()
+
+    input_pipeline_ids = automl.get_ensembler_input_pipelines(
+        _get_first_stacked_classifier_no() - 1
+    )
+    input_pipeline_ids.sort()
+
+    assert best_pipeline_ids == input_pipeline_ids
+
+    two_stacking_batches = 1 + 2 * (len(get_estimators(ProblemTypes.BINARY)) + 1)
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_batches=two_stacking_batches,
+        objective="Log Loss Binary",
+        ensembling=True,
+    )
+
+    test_env = AutoMLTestEnv("binary")
+    with test_env.test_context(mock_score_side_effect=score_side_effect):
+        automl.search()
+    pipeline_names = automl.rankings["pipeline_name"]
+    assert pipeline_names.str.contains("Ensemble").any()
+
+    ensemble_ids = [
+        _get_first_stacked_classifier_no() - 1,
+        len(automl.results["pipeline_results"]) - 1,
+    ]
+
+    final_best_pipeline_ids = [
+        pipeline["id"]
+        for pipeline in list(automl._automl_algorithm._best_pipeline_info.values())
+    ]
+    final_best_pipeline_ids.sort()
+
+    input_pipeline_0_ids = automl.get_ensembler_input_pipelines(ensemble_ids[0])
+    input_pipeline_0_ids.sort()
+
+    input_pipeline_1_ids = automl.get_ensembler_input_pipelines(ensemble_ids[1])
+    input_pipeline_1_ids.sort()
```

Comment:
Here, we're rerunning AutoMLSearch so that we can get two ensemble pipelines (and to make sure that the IDs for best pipeline are updated). I did this since there isn't a super easy way to verify that the first ensemble pipeline generated is correct. Def open to ideas/suggestions!

Reply:
Thoughts on rewriting [test_describe_pipeline_with_ensembling](https://github.com/alteryx/evalml/blob/main/evalml/tests/automl_tests/test_automl.py#L1469) to also call this new method. There what we did was check that the input pipeline ids for the second ensemble are all greater than the input ids of the first ensemble since the scores is always decreasing.

I'm ok with whatever you decide. This looks great to me as is!

Reply:
I am open to either, but I'm leaning slightly towards keeping it as-is. If we need to update `get_ensembler_input_pipelines()` in the future there's a single point to update + it keeps this particular edge case together. The runtime difference for another AutoMLSearch is relatively minimal for this mocked example.